# Getting started with OpenCV4

In [1]:
# Press CTRL + ENTER to run this line
# You should see an * between the [ ] on the left
# OpenCV takes a couple seconds to import the first time

import cv2

ModuleNotFoundError: No module named 'cv2'

In [5]:
# Let's verify the version of OpenCV you're running

print(cv2.__version__)

4.1.2


In [6]:
# Now let's import numpy
# We use as np, so that everything we call on numpy, we can type np instead
# It's short and looks neater and also the convention used

import numpy as np 

# Numpy is NumPy is a library for adding support for large, multi-dimensional 
# arrays and matrices, along with a large collection of high-level mathematical 
# functions to operate on these arrays.

In [7]:
# We don't need to do this again, but it's a good habit
import cv2 

# Load an image using 'imread' specifying the path to image
image = cv2.imread('./images/input.jpg')

# Our file 'input.jpg' is now loaded and stored in python 
# as a varaible we named 'image'

# To display our image variable, we use 'cv2.imshow'
# The first parameter will be title shown on image window
# The second parameter is the image varialbe
cv2.imshow('Loading Your First Image', image)

# 'waitKey' allows us to input information when a image window is open
# By leaving it blank it just waits for anykey to be pressed before 
# continuing. By placing numbers (except 0), we can specify a delay for
# how long you keep the window open (time is in milliseconds here)
cv2.waitKey()

# This closes all open windows 
# Failure to place this will cause your program to hang
cv2.destroyAllWindows()

In [8]:
# Same as above without the extraneous comments

import cv2 

image = cv2.imread('./images/input.jpg')

cv2.imshow('Loading Your First Image', image)
cv2.waitKey(0) # Notice 0 acts the same as leaving it blank like ()
cv2.destroyAllWindows()

### Let's take a closer look at how images are stored

In [11]:
# Import numpy
import numpy as np

print(image.shape)

(597, 968, 3)


### Shape gives the dimensions of the image array
The 2D dimensions are 597 pixels in high bv 968 pixels wide. The '3L' means that there are 3 other components (RGB) that make up this image.

In [12]:
# Let's print each dimension of the image

print('Height of Image:', int(image.shape[0]), 'pixels')
print('Width of Image: ', int(image.shape[1]), 'pixels')
print('Depth of Image: ', int(image.shape[2]), 'colors components')

Height of Image: 597 pixels
Width of Image:  968 pixels
Depth of Image:  3 colors components


### Saving images in OpenCV

In [13]:
# Simply use 'imwrite' specificing the file name and the image to be saved
cv2.imwrite('output.jpg', image)

True

In [14]:
# Or save it as a PNG (Portable Network Graphics) which is a Lossless bitmap image format
cv2.imwrite('output.png', image)

True